File: Shared/ChartThreshold.m
//
//  ChartThreshold.m
//  StockList Demo for iOS
//
//  Created by Gianluca Bertani on 26/02/14.
//  Copyright (c) Lightstreamer Srl
//

#import "ChartThreshold.h"
#import "ChartView.h"


@implementation ChartThreshold


#pragma mark -
#pragma mark Initialization

- (id) initWithView:(ChartView *)chartView value:(float)value {
	if ((self = [super init])) {

		// Initialization
		_chartView= chartView;

		_value= value;
	}
	
	return self;
}


#pragma mark -
#pragma mark Deletion

- (void) remove {
	[_chartView removeThreshold:self];
}


#pragma mark -
#pragma mark Properties

@synthesize chartView= _chartView;

@dynamic value;

- (float) value {
	return _value;
}

- (void) setValue:(float)value {
	_value= value;
	
	[_chartView setNeedsDisplay];
}

@synthesize mpnSubscription= _mpnSubscription;


@end

File: Shared/ChartThreshold.swift
//  Converted to Swift 5.4 by Swiftify v5.4.24488 - https://swiftify.com/
//
//  ChartThreshold.swift
//  StockList Demo for iOS
//
//  Created by Gianluca Bertani on 26/02/14.
//  Copyright (c) Lightstreamer Srl
//

import Foundation
import LightstreamerClient

class ChartThreshold: NSObject {
    // MARK: -
    // MARK: Properties
    private(set) var chartView: ChartView?

    private var _value: Float = 0.0
    @objc var value: Float {
        get {
            return _value
        }
        set(value) {
            _value = value

            chartView?.setNeedsDisplay()
        }
    }
    var mpnSubscription: MPNSubscription?

    // MARK: -
    // MARK: Initialization
    init(view chartView: ChartView?, value: Float) {
        super.init()
            // Initialization
            self.chartView = chartView

            self.value = value
    }

    // MARK: -
    // MARK: Removal

    // MARK: -
    // MARK: Deletion

    @objc func remove() {
        chartView?.remove(self)
    }

    // MARK: -
    // MARK: Properties
}
